pkg/sdkserver/sdkserver.go

```diff
@@ -105,6 +124,7 @@ type SDKServer struct {
 	gsReserveDuration   *time.Duration
 	gsPlayerCapacity    int64
 	gsConnectedPlayers  []string
+	gsCounterUpdates    map[string]counterUpdateRequest
```

Comment:
```suggestion
	gsCounterUpdates    map[string][]counterUpdateRequest
```

See below on tracking generation against each request, so we know if it's been applied against the GameServer resource we know about at any given moment during the workflow.

Somewhere we would need to do some cleanup, but this is the general gist.

Digging into some of the later code, maybe it's easier to do something more like:

```suggestion
	gsActiveCounterUpdates    map[string]counterUpdateRequest
	gsProcessedCounterUpdates    map[string][]counterUpdateRequest
```

So it's easier to know what's yes to be processed, and what may end up being needed when doing `get` requests from the SDK.

Reply:
Using gsCopy instead (see conversation in GetCounter).